src/common/Page.ts

```diff
@@ -1052,6 +1052,14 @@ export class Page extends EventEmitter {
       throw new Error(
         `Failed to add page binding with name ${name}: window['${name}'] already exists!`
       );
+
+    if (
+      typeof puppeteerFunction !== 'function' &&
+      typeof puppeteerFunction['default'] === 'function'
+    ) {
+      puppeteerFunction = puppeteerFunction['default'];
```

Comment:
```suggestion
      typeof puppeteerFunction.default === 'function'
    ) {
      puppeteerFunction = puppeteerFunction.default;
```

Reply:
Hey @mathiasbynens , thanks for your comments and suggestions.

I've added the test and is passing for me. Had to use ignore next line as it's expecting a function rather than an object:

```
    it('should fallback to default export when passed a module object', async () => {
      const { page, server } = getTestState();
      await page.goto(server.EMPTY_PAGE);
      var moduleObject = {
        default: function(a, b) {
                  return a * b;
                }
              };
      // @ts-expect-error
      await page.exposeFunction('compute', moduleObject);
      const result = await page.evaluate(async function () {
        return await globalThis.compute(9, 4);
      });
      expect(result).toBe(36);
    });
```



The changes to dot syntax in Page.ts is throwing an error for me though, please let me know if you prefer to ignore the line or keep the [ ] syntax.

```
src/common/Page.ts:1058:32 - error TS2339: Property 'default' does not exist on type 'never'.

1058       typeof puppeteerFunction.default === 'function'
```